src/experimentClient.ts

```diff
@@ -140,6 +140,7 @@ export class ExperimentClient implements Client {
     void this.flags.load();
     // eslint-disable-next-line no-void
     void this.variants.load();
+    this.mergeInitialFlagsWithStorage();
```

Comment:
this is a bit awkward, since the flags loading is async, so we add the flags to the cache, then the load immediately overwrites them.

I challenge you to find a good solution for this problem. Let me know and we can chat more about it.